crawler_mcp/crawlers/directory.py

```
@@ -415,27 +382,142 @@ async def _score_files(
+                for fut in asyncio.as_completed(futures):
+                    try:
+                        result = await fut
+                        batch_results.append(result)
+                        completed_count += 1
+
+                        # Check memory pressure every 16 files or at end of batch
+                        if completed_count % 16 == 0 or completed_count == len(futures):
+                            if (
+                                self.memory_manager
+                                and await self.memory_manager.check_memory_pressure()
+                            ):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)